frontend/src/Pages/HelpfulLinksManagement.tsx

```diff
@@ -48,20 +52,20 @@ export default function HelpfulLinksManagement() {
     const helpfulLinks = data?.data.helpful_links ?? [];
     const meta = data?.data.meta;
 
-    function updateLinks() {
-        addLinkModal.current?.close();
-        editLinkModal.current?.close();
-        void mutate();
-    }
-    async function deleteLink(id: number | undefined) {
-        const response = await API.delete(`helpful-links/${id}`);
-        if (response.success) {
-            updateLinks();
-            toaster('Helpful link deleted successfully', ToastState.success);
-        } else {
+    async function deleteLink() {
+        const response = await API.delete(`helpful-links/${currentLink?.id}`);
+        if (!response.success) {
             toaster('Error deleting helpful link', ToastState.error);
         }
-        deleteLinkModal.current?.close();
+        toaster('Helpful link deleted successfully', ToastState.success);
+        await mutate();
+        closeModal(deleteLinkModal);
+        setCurrentLink(null);
+    }
```

Comment:
Do you want to update this one to use the new checkResponse hook that you created?